controllers/valssecret_controller.go

```diff
@@ -17,9 +17,11 @@ limitations under the License.
 package controllers
 
 import (
+	"bytes"
 	"context"
 	b64 "encoding/base64"
 	"fmt"
+	"html/template"
```

Comment:
You need to use text/template here rather than html/template

Reply:
It's not a mistake, it's required by `sprig.FuncMap()`